core/src/main/java/com/netflix/conductor/core/listener/WorkflowStatusListener.java

```diff
@@ -22,4 +22,8 @@
     void onWorkflowCompleted(Workflow workflow);
 
     void onWorkflowTerminated(Workflow workflow);
+
+    default void onWorkflowFinalized(Workflow workflow) {
```

Comment:
General comment for all methods on this interface. The methods here are not called if `isWorkflowStatusListenerEnabled()` returns `false`. In every place where these need to be called, a conditional `if` block precedes. I suggest that we bring that check into the contract of this interface, like this,

```java
default void onWorkflowCompletedIfEnabled(Workflow workflow) {        if(workflow.getWorkflowDefinition().isWorkflowStatusListenerEnabled()) {
            onWorkflowCompleted(workflow);
        }
    }
```
A method like this can be added for all the listeners. It would make the code more readable.

Reply:
good idea, updated the interface